src/diffusers/pipeline_utils.py

```diff
@@ -65,7 +66,9 @@ def register_modules(self, **kwargs):
             # check if the module is a pipeline module
             pipeline_file = module.__module__.split(".")[-1]
             pipeline_dir = module.__module__.split(".")[-2]
-            is_pipeline_module = pipeline_file == "pipeline_" + pipeline_dir and hasattr(pipelines, pipeline_dir)
+            # is_pipeline_module = pipeline_file == "pipeline_" + pipeline_dir and hasattr(pipelines, pipeline_dir)
+            path = module.__module__.split(".")
+            is_pipeline_module = pipeline_dir in path and hasattr(pipelines, pipeline_dir)
```

Comment:
More general logic to detect if a module comes from `pipeline` module. For now this is only needed for `LDMBert` model and the safety checker.

This should probably be in a separate PR.